src/Diagnostics.RuntimeHost/Services/CacheService/TableCacheService.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Collections.Generic;
+using Diagnostics.RuntimeHost.Services.CacheService.Interfaces;
+using System.Collections.Concurrent;
+using Diagnostics.ModelsAndUtils.Models;
+using Diagnostics.RuntimeHost.Services.StorageService;
+using System.Threading.Tasks;
+using Diagnostics.ModelsAndUtils.Models.Storage;
+using Diagnostics.RuntimeHost.Models;
+using System.Linq;
+using System;
+
+namespace Diagnostics.RuntimeHost.Services.CacheService
+{
+    public class TableCacheService : ITableCacheService
+    {
+        private ConcurrentDictionary<string, List<DiagEntity>> cache;
+
+        private IStorageService storageService;
+
+        public TableCacheService(IStorageService service)
+        {
+            cache = new ConcurrentDictionary<string, List<DiagEntity>>();
+            storageService = service;
+        }
+        public void AddOrUpdate(string key, List<DiagEntity> value)
+        {
+            cache.AddOrUpdate(key, value, (existingKey, existingValue) => value);
+        }
+
+        public bool ContainsKey(string key)
+        {
+            return cache.ContainsKey(key);
+        }
+
+        public IEnumerable<List<DiagEntity>> GetAll()
+        {
+            return cache.Values;
+        }
+
+        public bool TryGetValue(string key, out List<DiagEntity> value)
+        {
+            return cache.TryGetValue(key, out value);
+        }
+
+        public bool TryRemoveValue(string key, out List<DiagEntity> value)
+        {
+            return cache.TryRemove(key, out value);
+        }
+
+        public async Task<List<DiagEntity>> GetEntityListByType<TResource>(RuntimeContext<TResource> context, string entityType = null)
+          where TResource : IResource
+        {
+            entityType = entityType == null ? "Detector" : entityType;
+
+            var result = new List<DiagEntity>();
+
+            while (!TryGetValue(entityType, out result))
```

Comment:
does this needs to be "if" instead of "while"

Reply:
Oh I see what you are doing. One other way to save little bit of computation is update line 59 to say : 
result = await storageService.RetieveEntitiesByPartitionkey(entityType);

Reply:
The reason I have this as while is, first time we are querying for list of entities, there won't be any in the cache. So it can get the entities from storage and add to cache. When the while loop executes second time,  the condition will be met so it can proceed to the next statement.